### PR TITLE
Set color as an ARGB int32 in the render buffer.

### PR DESCRIPTION
**Pull request by bendmorris**

### CODE REVIEW

**Comment by MattTuttle** [comment]
My only thought is that we might want to calculate color+alpha at a higher level. Backdrop and Tilemap would improve if they only had to calculate this once per group instead of per triangle. This could be done in a later pull request though.

File: haxepunk/graphics/shaders/Shader.hx [hunk 1]
```diff
@@ -107,69 +108,53 @@ class Shader
...
-			if (hasTexCoord)
-			{
-				buffer[++texturePos] = data.uvx1;
-				buffer[++texturePos] = data.uvy1;
-				buffer[++texturePos] = data.uvx2;
-				buffer[++texturePos] = data.uvy2;
-				buffer[++texturePos] = data.uvx3;
-				buffer[++texturePos] = data.uvy3;
-			}
+			var c:UInt = hasColor ? data.color.withAlpha(data.alpha) : 0;
```
Comment (MattTuttle) [inline haxepunk/graphics/shaders/Shader.hx]: We might want to calculate this at a higher level. Backdrop and Tilemap would benefit from this since they generate a bunch of triangles.

File: haxepunk/graphics/shaders/TextureShader.hx [hunk 2]
```diff
@@ -11,14 +11,14 @@ precision mediump float;
...
-attribute vec4 aColor;
 attribute vec2 aTexCoord;
+attribute vec4 aColor;
 varying vec2 vTexCoord;
 varying vec4 vColor;
 uniform mat4 uMatrix;
 
 void main(void) {
-	vColor = aColor;
+	vColor = vec4(aColor.bgr * aColor.a, aColor.a);
```
Comment (MattTuttle) [inline haxepunk/graphics/shaders/TextureShader.hx]: Nice GPU optimization!

File: haxepunk/graphics/atlas/RenderBuffer.hx [hunk 3]
```diff
@@ -0,0 +1,73 @@
+package haxepunk.graphics.atlas;
+
+#if js
+import js.html.Int32Array;
+#end
+import flash.gl.GL;
+import flash.gl.GLBuffer;
+
+class RenderBuffer
```
Comment (MattTuttle) [inline haxepunk/graphics/atlas/RenderBuffer.hx]: Great idea pulling the buffer into its own class.

File: haxepunk/graphics/shaders/Shader.hx (hunk 1 above)
Comment (bendmorris) [inline haxepunk/graphics/shaders/Shader.hx]: Good idea. We can do that in a separate commit.
